src/WingetCreateCLI/Commands/UpdateCommand.cs

```diff
@@ -199,42 +200,43 @@ public async Task<Manifests> DeserializeExistingManifestsAndUpdate(List<string>
                 return null;
             }
 
-            if (!PackageParser.UpdateInstallerNodesAsync(
-                installerManifest,
-                this.InstallerUrls,
-                packageFiles,
-                out bool installerMismatch,
-                out List<WingetCreateCore.Models.Installer.Installer> unmatchedInstallers,
-                out List<WingetCreateCore.Models.Installer.Installer> multipleMatchedInstallers,
-                out List<PackageParser.DetectedArch> detectedArchOfInstallers))
-            {
-                DisplayMismatchedArchitectures(detectedArchOfInstallers);
+            List<PackageParser.DetectedArch> detectedArchOfInstallers;
 
-                if (installerMismatch)
-                {
-                    Logger.ErrorLocalized(nameof(Resources.NewInstallerUrlMustMatchExisting_Message));
+            try
+            {
+                PackageParser.UpdateInstallerNodesAsync(
+                    installerManifest,
+                    this.InstallerUrls,
+                    packageFiles,
+                    out detectedArchOfInstallers);
+            }
+            catch (InvalidOperationException)
+            {
+                Logger.ErrorLocalized(nameof(Resources.InstallerCountMustMatch_Error));
+                return null;
+            }
+            catch (Exceptions.ParsePackageException parsePackageException)
+            {
+                var parseFailedInstallerUrls = parsePackageException.ParseFailedInstallerUrls;

```

Comment:
[nit] IMO, assigning this to a new variable here isn't worth it, and below

Reply:
remove assignment to new variable